### ISSUE TITLE
Domain Connection subscription removal page is confusing.

### ISSUE THREAD

**Issue by csabarakasz** [Automattic/wp-calypso#68344]
### What

[The user](37114048-hc) got confused by Domain Connection Sub. removal page.

Step one, you can tell that it is a domain connection sub (although it is not that highlighted):
![DErD3jvYE7UxggsgOMxGaQareTNRhGVlFp7HISo6.jpg](https://user-images.githubusercontent.com/84539773/192520364-74300fd3-0587-4792-9b1b-bd88e372562f.jpg)

When they click Remove, the next "Share your Feedback page" is a general page and can be confusing to know what product it this about. The only thing they see is "Remove Product [SITE URL]" which the user took that they are cancelling their site. 
![oYdZnzsZ56ktPsFkD7vnRn0B9gpA5mfNScJPVhC1.jpg](https://user-images.githubusercontent.com/84539773/192521125-6073cb7d-c219-4f7f-bfc6-fe583c8d1d67.jpg)

We could include a confirmation of the name of the product/s that are being removed. (Although I realize that this page is the same for all product removals, so this might be complicated.)

### Why

To avoid confusion. 

### How

Adding an extra line where we list what product is being removed.

**Comment by zachspears** [comment]
I agree with the potential confusion here since users commonly refer to the domain name when referring to their site. Marking low priority due to low impact.

**Comment by ccwalburn** [comment]
Another user was confused by this. They had two domains and were deleting the non-primary domain, but the messaging referred to the primary domain so they thought it would delete their entire site.

Since this is a low priority and could be complicated to make product-specific, could we at least ask the developers to change the verbiage to "Remove product **from**: SITE URL"?

36758180-hc
